### PR TITLE
Optimize jobs API by prefetching data

### PR DESCRIPTION
**Pull request by perlpunk**
For /jobs?... API requests, for each job found, another ~ 4 SQL queries
are executed.

Prefetching data will be faster (50% in my local tests).

Issue: https://progress.opensuse.org/issues/94111

### CODE REVIEW

**Review by kalikiana — APPROVED** [review]

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -499,21 +502,29 @@ sub to_hash {
...
         push @{$j->{assets}->{$_->type}}, $_->name for @$assets;
     }
     if ($args{deps}) {
-        $j = {%$j, %{$job->dependencies}};
+        if ($dependencies) {
+            $j = {%$j, %{$job->dependencies($dependencies->{children} || [], $dependencies->{parents} || [])}};
+        }
+        else {
+            $j = {%$j, %{$job->dependencies()}};
+        }
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/Jobs.pm:527]:
```suggestion
        $j = {%$j, $dependencies ? %{$job->dependencies($dependencies->{children} || [], $dependencies->{parents} || [])} : %{$job->dependencies()}};
```

but the amount of curly braces is hardly readable in both cases. Maybe better extract a function to return what is needed here

**Review by okurz — COMMENTED** [review]
Do you think it's feasible to add a test checking for the number of SQL queries conducted and ensuring that it's now less than in before?

**Review by Martchus — COMMENTED** [review]
Looks good. (I'm not approving to give you a chance to read @okurz 's comments before it would be merged automatically.)

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 1 above)
Comment (perlpunk) [reply lib/OpenQA/Schema/Result/Jobs.pm]: I rearranged it a bit

**Review by okurz — COMMENTED** [review]
Do you think it's feasible to add a test checking for the number of SQL queries conducted and ensuring that it's now less than in before?

**Review by Martchus — APPROVED** [review]

**Pull request merged**